include/boost/json/detail/impl/digest.ipp

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright (c) 2019 Vinnie Falco (vinnie.falco@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/cppalliance/json
+//
+
+#ifndef BOOST_JSON_DETAIL_IMPL_DIGEST_IPP
+#define BOOST_JSON_DETAIL_IMPL_DIGEST_IPP
+
+namespace boost {
+namespace json {
+namespace detail {
+
+inline
+std::uint32_t
+digest_impl(
+    char const* s,
+    std::size_t n,
+    std::size_t salt,
+    std::false_type) noexcept
+{
+    std::uint32_t const prime = 0x01000193UL;
+    std::uint32_t hash  = 0x811C9DC5UL;
+    hash += static_cast<std::uint32_t>(salt);
+    for(;n--;++s)
+        hash = (*s ^ hash) * prime;
+    return hash;
+}
+
+inline
+std::uint64_t
+digest_impl(
+    char const* s,
+    std::size_t n,
+    std::size_t salt,
+    std::true_type) noexcept
+{
+    std::uint64_t const prime = 0x100000001B3ULL;
```

Comment:
We could just use `BOOST_JSON_ARCH` to set the constants instead of having two overloads

Reply:
E.g. 
```cpp
#if BOOST_JSON_ARCH == 64
std::uint64_t const prime = 0x100000001B3ULL;
std::uint64_t hash  = 0xcbf29ce484222325ULL;
#else 
std::uint32_t const prime = 0x01000193UL;
std::uint32_t hash  = 0x811C9DC5UL;
#endif 
...
```